### PR TITLE
POO97076

### PR DESCRIPTION
**Pull request by vtrubovics**
NETFILTER_PKT test fix in ip_eb_tables

- Related ticket: https://bugzilla.suse.com/show_bug.cgi?id=1188851, https://progress.opensuse.org/issues/97076
- Gitlab merge request https://gitlab.suse.de/security/audit-test-sle15/-/merge_requests/25
- Verification run: openqa.mypersonalinstance.de/tests/xyz#step/module/x

### CODE REVIEW

File: tests/security/cc/ip_eb_tables.pm [hunk 1]
```diff
@@ -23,9 +23,34 @@ sub run {
 
     select_console 'root-console';
 
+    #Configure bridge for ip_eb_tables workaround
+    assert_script_run("echo \"BOOTPROTO='dhcp'\" > /etc/sysconfig/network/ifcfg-br0");
+    assert_script_run("echo \"STARTMODE='auto'\" >> /etc/sysconfig/network/ifcfg-br0");
+    assert_script_run("echo \"BRIDGE='yes'\" >> /etc/sysconfig/network/ifcfg-br0");
+    assert_script_run("echo \"BRIDGE_PORTS='eth0'\" >> /etc/sysconfig/network/ifcfg-br0");
+    assert_script_run("echo \"BRIDGE_STP='off'\" >> /etc/sysconfig/network/ifcfg-br0");
```
Comment (rfan1) [inline tests/security/cc/ip_eb_tables.pm:31]: Hi Viktor, 

I suggest you can use one command line to modify the file like below code:

"os-autoinst-distri-opensuse/tests /console/rsync_server.pm"

    script_run 'cat > /etc/rsyncd.conf <<EOF
max connections = 2
log file = /var/log/rsync.log
timeout = 300

[pub]
     comment = Testing files available for download
     read only = yes
     list = yes
     path = /srv/rsync_test/pub
     uid = nobody
     gid = nobody
     auth users = test42
     secrets file = /etc/rsyncd.secrets

EOF
true';

File: tests/security/cc/ip_eb_tables.pm [hunk 2]
```diff
@@ -23,9 +23,34 @@ sub run {
 
     select_console 'root-console';
 
+    #Configure bridge for ip_eb_tables workaround
```
Comment (lilyeyes) [inline tests/security/cc/ip_eb_tables.pm:26]: One blank space needed here `s/#C/# C/`

File: tests/security/cc/ip_eb_tables.pm [hunk 3]
```diff
@@ -23,9 +23,34 @@ sub run {
...
+    assert_script_run("echo \"BRIDGE_STP='off'\" >> /etc/sysconfig/network/ifcfg-br0");
+    assert_script_run("echo \"BRIDGE_FORWARDDELAY='15'\" >> /etc/sysconfig/network/ifcfg-br0");
+    
+    # Backup eth0 configuration
+    assert_script_run("cp /etc/sysconfig/network/ifcfg-eth0 /etc/sysconfig/network/ifcfg-eth0.bak");
+
+    # eth0 coniguration
+    assert_script_run("echo \"IPADDR='0.0.0.0'\" > /etc/sysconfig/network/ifcfg-eth0");
+    assert_script_run("echo \"BOOTPROTO='none'\" >> /etc/sysconfig/network/ifcfg-eth0");
+    assert_script_run("echo \"STARTMODE='auto'\" >> /etc/sysconfig/network/ifcfg-eth0");
```
Comment (lilyeyes) [inline tests/security/cc/ip_eb_tables.pm:40]: Could you please use a variable to invoke `/etc/sysconfig/network/ifcfg-eth0` and `/etc/sysconfig/network/ifcfg-br0`?
e.g.,
my $f_ifcfg-br0 = '/etc/sysconfig/network/ifcfg-br0';
assert_script_run("echo \"BOOTPROTO='dhcp'\" > $$f_ifcfg-br0 ");

**Comment by lilyeyes** [comment]
Please revise these lines according to prompt in line:
[type description here, PLEASE, REMOVE THIS LINE, PLACEHOLDER, BEFORE SUBMITTING THIS PULL REQUEST]

    Related ticket: https://progress.opensuse.org/issues/xyz
    Needles: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/xyz
    Verification run: openqa.mypersonalinstance.de/tests/xyz#step/module/x

**Comment by lilyeyes** [comment]
Please revise the PR's commit message according to:
https://chris.beams.io/posts/git-commit/
or,
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md
In short (The seven rules of a great Git commit message):  
1.   Separate subject from body with a blank line
2.     Limit the subject line to 50 characters
3.     Capitalize the subject line
4.     Do not end the subject line with a period
5.     Use the imperative mood in the subject line
6.     Wrap the body at 72 characters
7.     Use the body to explain what and why vs. how

**Review by Amrysliu — COMMENTED** [review]
The commit message title should be a summary of the purpose.
There are two commits in this pull request, but I didn't see a big difference between them. Maybe one commit should be ok.
CI check fails due to tidy, please check the code style.

**Comment by rfan1** [comment]
> The commit message title should be a summary of the purpose.
> There are two commits in this pull request, but I didn't see a big difference between them. Maybe one commit should be ok.
> CI check fails due to tidy, please check the code style.

perltidy -l=160 -fbl -nsfs -baao -bbao -pt=2 -bt=2 -sbt=2 -sct -o /tmp/ooo your_test.pm

**Comment by jouyingbin** [comment]
1. Please describe more detail in the git commit message and GitHub description.
2. Please fill in the detailed links using the template format.
```
    Related ticket: https://progress.opensuse.org/issues/xyz
    Needles: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/xyz
    Verification run: openqa.mypersonalinstance.de/tests/xyz#step/module/x
```

File: tests/security/cc/ip_eb_tables.pm (hunk 3 above)
Comment (vtrubovics) [reply tests/security/cc/ip_eb_tables.pm]: Fixed

File: tests/security/cc/ip_eb_tables.pm (hunk 2 above)
Comment (vtrubovics) [reply tests/security/cc/ip_eb_tables.pm]: Fixed

File: tests/security/cc/ip_eb_tables.pm (hunk 1 above)
Comment (vtrubovics) [reply tests/security/cc/ip_eb_tables.pm]: Fixed

**Comment by lilyeyes** [comment]
The changes look good now.
But I can not access your verification run link.

**Comment by lilyeyes** [comment]
I triggered cc_auidt-test which contains ipusing your branch and let's see the run:
```# /usr/share/openqa/script/client isos post --host http://openqa.nue.suse.com _GROUP_ID=168 TEST=cc_audit-test DISTRI=sle VERSION=15-SP3  ARCH=x86_64  BUILD=187.1 FLAVOR=Online _SKIP_CHAINED_DEPS=1  CASEDIR=https://github.com/vtrubovics/os-autoinst-distri-opensuse/#POO97076 --apikey 138AE0DD066521BA --apisecret C5841DF75DE37BAE CODE_BASE=https://gitlab.suse.de/viktors.trubovics/audit-test-sle15/-/archive/POO97076/audit-test-sle15-POO97076.tar
{ count => 1, failed => [], ids => [6929096], scheduled_product_id => 783471 }
```

Please check the running of: https://openqa.suse.de/tests/6929096

**Comment by lilyeyes** [comment]
> NETFILTER_PKT test fix in ip_eb_tables
> 
>     * Related ticket: https://bugzilla.suse.com/show_bug.cgi?id=1188851, https://progress.opensuse.org/issues/97076
> 
>     * Verification run: openqa.mypersonalinstance.de/tests/xyz#step/module/x

Please add the MR link as well, thanks :)

**Review by lilyeyes — APPROVED** [review]
LGTM. Thanks for the changes

**Comment by lilyeyes** [comment]
Though I could not access the VR but talked offline the code changes work.

**Comment by rfan1** [comment]
Hi Viktor,

I suggest you change the PR's commit message from "POO97076" to "NETFILTER_PKT test fix in ip_eb_tables"

**Review by Amrysliu — APPROVED** [review]
LGTM

**Comment by Amrysliu** [comment]
Please check the CI check, it failed on `tidy`.
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13112/checks?check_run_id=3501893250

**Review by Amrysliu — CHANGES_REQUESTED** [review]
Please fix the CI failure.

**Comment by vtrubovics** [comment]
> 
> 
> > The commit message title should be a summary of the purpose.
> > There are two commits in this pull request, but I didn't see a big difference between them. Maybe one commit should be ok.
> > CI check fails due to tidy, please check the code style.
> 
> perltidy -l=160 -fbl -nsfs -baao -bbao -pt=2 -bt=2 -sbt=2 -sct -o /tmp/ooo your_test.pm

**Pull request closed without merging**